frontend/scripts/generateSwaggerClient.ts

```diff
@@ -0,0 +1,39 @@
+import { exec } from 'child_process';
+import 'dotenv/config';
+import * as fs from 'fs';
+import path from 'path';
+import { exit } from 'process';
+
+const outDir = path.join(__dirname, '..', 'src/api/client');
+const tempDir = path.join(__dirname, '..', '.openapi-temp');
+
+// Make the directory if it doesn't exist
+if (!fs.existsSync(outDir)) {
+  fs.mkdirSync(outDir);
+}
+
+const specURL = './swagger.yml';
+
+console.log(`Generating client API from spec at ${specURL}`);
+
+// Generate the client api
+exec(
+  `openapi-generator-cli generate -g typescript-axios --additional-properties=useSingleRequestParameter=true,withoutPrefixEnums=true, -i ${specURL} -o ${tempDir}`,
```

Comment:
Could we do this by just putting the shell script directly as a script in package.json?

Reply:
Could technically do that but itd get kinda ugly with all the other operations like creating and deleting folders and also cant have one solution thatll work on all Operating Systems